build

```diff
@@ -1042,65 +1042,116 @@ def build_next ():
 
 
 
+def version_from_git (folder):
+  try:
+    log ('fetching version information from git in folder "' + folder + '": ')
+    with open (os.devnull) as devnull:
+      tag = subprocess.check_output ([ 'git', 'describe', '--abbrev=0' ], cwd=folder, stderr=devnull).decode(errors='ignore').strip()
+      commit = subprocess.check_output ([ 'git', 'describe', '--abbrev=8', '--dirty', '--always' ], cwd=folder, stderr=devnull).decode(errors='ignore').strip()
+    log (tag + ', ' + commit + '\n')
+    return [ tag, commit ]
+  except (OSError, subprocess.CalledProcessError):
+    log ('not found\n')
+    raise LookupError
+
+
+
+def version_from_file (version_file, version_macro):
+  log ('fetching version information from "' + version_macro + '" in "' + version_file + '": ')
+  try:
+    with open (version_file) as fd:
+      for line in fd:
+        line = line.split ("//")[0]
+        if version_macro in line:
+          tag = line.split(version_macro)[1].strip().strip('"')
+          log (tag + '\n')
+          return tag
+    if not len (tag):
+      raise NameError
+  except (OSError, NameError):
+    log ('not found\n')
+    raise LookupError
+
+
 
-def get_git_version (folder):
+
+def mrtrix_version ():
   try:
-    return get_git_version.version
+    return mrtrix_version.version
   except AttributeError:
     pass
 
   log ('''
-getting git version in folder "''' + folder + '"...\n')
+getting MRtrix version from folder "''' + mrtrix_dir[-1] + '"...\n  ')
+  try:
+    tag = version_from_file (os.path.join (lib_dir, 'version.h'), 'MRTRIX_BASE_VERSION')
+  except LookupError:
+    fail ('ERROR: failed to read version information from file "' + os.path.join (lib_dir, 'version.h') + '"\n')
 
-  log ('  reading from core/version.h: ')
   try:
-    with open (os.path.join (lib_dir, 'version.h')) as fd:
-      for line in fd:
-        line = line.split ("//")[0]
-        if 'MRTRIX_BASE_VERSION' in line:
-          hard_coded_version = line.split('MRTRIX_BASE_VERSION')[1].strip().strip('"')
-          log (hard_coded_version + '\n')
-    if not len (hard_coded_version):
-      raise NameError
-  except ( OSError, NameError ):
-    fail ("ERROR: failed to read version information\n")
+    log ('  ')
+    version = version_from_git (mrtrix_dir[-1])
+    if version[0] != tag:
+        fail ('ERROR: version stored in "' + os.path.join (lib_dir, 'version.h') + '" does not match git tag name\n')
```

Comment:
This trips up run_tests when updating from an older version as testing/src/project_version.h does not seem to be cleared before running the tests. The version string in that file was 3.0_RC2-931-g3688ac7a. Is this a one-off?

Reply:
Yes, that file is not under version control, and used to be generated by the `build` script. This new version will no longer update that file, but if present, check that it matches the nearest tag (I think...). So it should just be a matter of deleting that file...

Reply:
> This trips up run_tests when updating from an older version as testing/src/project_version.h does not seem to be cleared before running the tests.

Yep, I ran into the same issue yesterday.